crates/sui-types/src/passkey_authenticator.rs

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use crate::crypto::PublicKey;
+use crate::crypto::Secp256r1SuiSignature;
+use crate::crypto::SuiSignatureInner;
+use crate::signature_verification::VerifiedDigestCache;
+use crate::{
+    base_types::{EpochId, SuiAddress},
+    crypto::{DefaultHash, Signature, SignatureScheme, SuiSignature},
+    digests::ZKLoginInputsDigest,
+    error::{SuiError, SuiResult},
+    signature::{AuthenticatorTrait, VerifyParams},
+};
+use fastcrypto::hash::{HashFunction, Sha256};
+use fastcrypto::rsa::{Base64UrlUnpadded, Encoding};
+use fastcrypto::secp256r1::{Secp256r1PublicKey, Secp256r1Signature};
+use fastcrypto::traits::VerifyingKey;
+use fastcrypto::{error::FastCryptoError, traits::ToFromBytes};
+use once_cell::sync::OnceCell;
+use passkey::types::webauthn::{ClientDataType, CollectedClientData};
```

Comment:
Question: Given all we're using the passkey library for is these data types could we either:
1. only depend on the passkey_types crate and not the whole passkey top level crate
2. or can we just reimplement these types here so we don't need to depend on them at all?

I think 1 is probably the safest option and we could start there.

Reply:
revised, only passkey-types is included in deps. passkey-authenticator and passkey-client are included for dev-deps for e2e tests and unit tests